lib/reline/line_editor.rb

```diff
@@ -935,37 +935,23 @@ def dialog_proc_scope_completion_journey_data
   end
 
   private def run_for_operators(key, method_symbol, &block)
-    if @waiting_operator_proc
+    if @vi_waiting_operator
       if VI_MOTIONS.include?(method_symbol)
         old_byte_pointer = @byte_pointer
-        @vi_arg = @waiting_operator_vi_arg if @waiting_operator_vi_arg&.> 1
+        @vi_arg = (@vi_arg || 1) * @vi_waiting_operator_arg
         block.(true)
         unless @waiting_proc
           byte_pointer_diff = @byte_pointer - old_byte_pointer
           @byte_pointer = old_byte_pointer
-          @waiting_operator_proc.(byte_pointer_diff)
-        else
-          old_waiting_proc = @waiting_proc
-          old_waiting_operator_proc = @waiting_operator_proc
-          current_waiting_operator_proc = @waiting_operator_proc
-          @waiting_proc = proc { |k|
-            old_byte_pointer = @byte_pointer
-            old_waiting_proc.(k)
-            byte_pointer_diff = @byte_pointer - old_byte_pointer
-            @byte_pointer = old_byte_pointer
-            current_waiting_operator_proc.(byte_pointer_diff)
-            @waiting_operator_proc = old_waiting_operator_proc
-          }
+          send(@vi_waiting_operator, byte_pointer_diff)
```

Comment:
Yes. multi-byte characters are allowed.

vi_waiting_operator will do
1. save cursor position (represented by `@byte_pointer`)
2. accept VI_MOTIONS. cursor position moves. (`@line_index` does not change)
3. Run an action(==`@vi_waiting_operator`) to a text between old cursor position and new cursor position

Current cursor position can be retrieved from `@byte_pointer`, so we need to pass `old cursor_position` or `difference between old and new curspor_position`
The original implementation selected `diff`, and this pull request follows it.

And we can choose how to represent cursor position: `byte_pointer`, `string size`, `grapheme_cluster size` `width in column when rendered to terminal`. Using `byte_position` is easiest because we don't need to translate `@byte_pointer` to another representation form.

Reply:
I understood. Thanks for explaining!